### PR TITLE
Gh1150 docstrip

### PR DESCRIPTION
**Pull request by FrankMittelbach**
# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [n/a] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [n/a] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: base/doc/ltnews41.tex [hunk 1]
```diff
@@ -370,6 +370,18 @@ \subsection{Fix the use of  \texttt{localmathalpabets}}
 %
 \githubissue[s]{1101 1028}
 
+\subsection{\pkg{docstrip}:\ Error if \texttt{.ins} file is problematical}
+
+If a file to generate had the same name as a preamble declared with
+\cs{declarepreamble} the preamble definition was overwritten because
+the macro used to stored it was reused for denoting the output
```
Comment (muzimuzhi) [inline base/doc/ltnews41.tex:377]:
```suggestion
the macro used to store it was reused for denoting the output
```

File: base/docstrip.dtx [hunk 2]
```diff
@@ -3027,8 +3027,37 @@ Z
 %    defining preamble for the current file.
 %    \begin{macrocode}
 \def\@fileX#1#2{%
+%    \end{macrocode}
+%    If the csname used for the stream has already been defined, e.g.,
+%    as a preamble or postamble or for some other purpose, chances are
```
Comment (muzimuzhi) [inline base/docstrip.dtx:3032]:
```suggestion
%    as a preamble or postamble or for some other purposes, chances are
```

File: base/docstrip.dtx [hunk 3]
```diff
@@ -3027,8 +3027,37 @@ Z
...
+%    \end{macrocode}
+%    If it matches the name of the current preamble or postable then
+%    it definitely can't work, so we call that out explicitly:
+%    \begin{macrocode}
+  \ifx\curout\currentpreamble
+     \errmessage{Declared preamble name \string#1 not allowed if
+       \string\file{\@stripstring#1} is used}%
+  \fi
+  \ifx\curout\currentpostamble
+       \errmessage{Declared postamble name \string#1 not allowed if
```
Comment (muzimuzhi) [inline base/docstrip.dtx:3055]: Reduce indent level by 1.
```suggestion
     \errmessage{Declared postamble name \string#1 not allowed if
```

**Review by muzimuzhi — CHANGES_REQUESTED** [review]
Typos and minor improvements.

**Review by josephwright — APPROVED** [review]

File: base/docstrip.dtx [hunk 4]
```diff
@@ -3027,8 +3027,37 @@ Z
...
+  \ifx#1\relax \else
+     \errmessage{Command \string#1 for denoting the output \noexpand
+       \file stream already defined!^^J
+       \space Current meaning is:^^J^^J\meaning#1^^J^^J
+       \space Extraction will probably fail - check result}%
+  \fi
   \chardef#1=\stream@closed
   \def\curout{#1}%
+%    \end{macrocode}
+%    If it matches the name of the current preamble or postable then
```
Comment (muzimuzhi) [inline base/docstrip.dtx:3047]:
```suggestion
%    If it matches the name of the current preamble or postamble then
```

**Review by muzimuzhi — APPROVED** [review]
one spelling

File: base/docstrip.dtx (hunk 4 above)
Comment (FrankMittelbach) [reply base/docstrip.dtx:3047]: thanks, as usual

**Pull request merged**